### PR TITLE
feat(datasets): add dataset related functions

### PR DESCRIPTION
**Pull request by jinglinpeng**
# Description

1. polish readme.
2. add load_dataset and get_dataset_names

# How Has This Been Tested?

covered by test case.

image

### CODE REVIEW

File: dataprep/datasets/_base.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
+"""This module implements load dataset related functions"""
```
Comment (dovahcrow) [inline dataprep/datasets/_base.py:1]: Why is the file underscored?

File: dataprep/datasets/_base.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+import os
+from os.path import dirname
+from typing import List
+
+import pandas as pd
+
+
+def get_dataset_names() -> List[str]:
```
Comment (dovahcrow) [inline dataprep/datasets/_base.py:10]: Are we supposed to put the data in https://github.com/sfu-db/dataprep-data?

Comment (jinglinpeng) [reply dataprep/datasets/_base.py:10]: No. The data set is stored in the same repo, under `data` folder.

File: dataprep/datasets/_base.py (hunk 1 above)
Comment (jinglinpeng) [reply dataprep/datasets/_base.py:1]: It follows the file name of sklearn. `https://github.com/scikit-learn/scikit-learn/tree/master/sklearn/datasets`. I can remove the underscore if you think it's better.

**Review by dovahcrow — APPROVED** [review]

**Comment by dovahcrow** [comment]
fixes #405 

@jinglinpeng It would be better to mention the related issue in the PR so that they can be auto closed.

**Pull request merged**